config/v2/resolvers.go

```diff
@@ -99,6 +100,21 @@ func ResolveAWSProvider(commons ...Common) *AWSProvider {
 	return nil
 }
 
+func ResolveSnowflakeProvider(commons ...Common) *SnowflakeProvider {
+	account := lastNonNil(SnowflakeProviderAccountGetter, commons...)
+	role := lastNonNil(SnowflakeProviderRoleGetter, commons...)
+	region := lastNonNil(SnowflakeProviderRegionGetter, commons...)
+
+	if account != nil || role != nil || region != nil {
```

Comment:
below we make the assumption that if one is set then all are

Reply:
You mean below in this func? I don't think that's the case.

The current design thinking here is that the "resolution" logic will return the best version of the data we have, though it might be incomplete. Then validation will ensure that it is valid. Once we ensure that its valid we can confer the configs to plans knowing that all preconditions are met.